rules.d/60-sensor.rules

```diff
@@ -4,7 +4,7 @@ ACTION=="remove", GOTO="sensor_end"
 
 # device matching the sensor's name and the machine's DMI data for IIO devices
 SUBSYSTEM=="iio", KERNEL=="iio*", SUBSYSTEMS=="usb|i2c|platform", \
-  IMPORT{builtin}="hwdb 'sensor:modalias:$attr{modalias}:$attr{[dmi/id]modalias}'", \
+  IMPORT{builtin}="hwdb 'sensor:modalias:$attr{modalias}:$attr{[dmi/id]modalias}:$attr{[dmi/id]product_sku}'", \
```

Comment:
So I guess that the addition of the ':$attr{[dmi/id]product_sku}' part here is to match on the :0A3E* at the end of the Latitude 9520 match ?

Changing the match passed to a hwdb import like this is not a good idea, as it may very well break other matches in the hwdb. This change is only safe if **all** current matches in the hwdb end with an asterisk ('*'). Checking 60-sensor.hwdb I 've found the first matches where this is not the case at machines whose vendor starts with a C (Chuwi HiBook entries), I assume there will be more if I check the entire file. So I'm afraid that adding the ':$attr{[dmi/id]product_sku}' at the end here is not a good idea. Also the product_sku DMI string is a recent(ish) addition to the DMI spec and not all devices will have this set. I wonder what udev will do here when there is $attr{[dmi/id]product_sku} does not exist.

May I ask why matching on the SKU is necessary here? Are other 9520 SKUs 2-in-1s or do they not have the sensor ?  If the other SKUs don't have the sensor then you can simply drop the SKU part from the match, since then the sensor part of the match then will fail anyways.  If some of the other SKUs actually arr 2-in-1s then we need to figure something out here.

Reply:
Indeed, I found some line without asterisk('*'). Thanks for looking into it carefully.

The Latittude 9520 has two SKUs clamshell and 2-in-1, both of them have sensor, but we intend to disable rotation for clamshell model. Currently, I don't 2-in-1 model in hand, so I am not sure if modalias is enough to descrminate. As you said, to add product sku is not a good idea. I'll request Dell to provide more info for knowing how to handle it better. Sorry for wasting your time.

Reply:
https://github.com/systemd/systemd/pull/17358 adds the trailing stars. We should do that for future expansibility, irrespectively of this PR.